sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,310 @@
+  archived: "archived",
+  recordMatch: "Record match",
+  record: "The record was not found",
+};
+
+/**
+ * Possible warnings unique identifiers.
+ * !These keys are also consumed in the UI to display/hide warning banners.
+ */
+export enum ApplicationWithdrawalValidationWarnings {
+  ApplicationNotFound = "applicationNotFound",
+  NoMatchingRecordFound = "noMatchingRecordFound",
+  HasPreviouslyBeenWithdrawn = "hasPreviouslyBeenWithdrawn",
+}
+
+/**
+ * Types of non-critical errors that will not prevent
+ * the offering from being saved.
+ */
+export enum ValidationContextTypes {
+  Warning = "warning",
+}
+
+/**
+ * Represent the context of an error that has an additional context to
+ * express a possible warning or some relevant information captured during
+ * the validation process.
+ * All validations when failed will generate an error. The ones that have
+ * the warning or information contexts will not be considered critical.
+ */
+export class ValidationContext {
+  /**
+   * Creates an error context that will make the error downgrade to
+   * a condition of a warning information will force the offering
+   * to be reviewed by the Ministry.
+   * @param warningTypeCode warning code that uniquely identifies this condition.
+   * @returns the warning context.
+   */
+  static CreateWarning(
+    warningTypeCode: ApplicationWithdrawalValidationWarnings,
+  ): ValidationContext {
+    const newContext = new ValidationContext();
+    newContext.type = ValidationContextTypes.Warning;
+    newContext.typeCode = warningTypeCode;
+    return newContext;
+  }
+
+  /**
+   * Context type.
+   */
+  type: ValidationContextTypes;
+  /**
+   * Unique identifier of the validation error.
+   */
+  typeCode: ApplicationWithdrawalValidationWarnings;
+}
+
+export class ApplicationBulkWithdrawalImportBusinessValidationModel {
+  /**
+   * Data record type.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @Equals(RecordType.ApplicationBulkWithdrawalDataRecordType, {
+    message: `${DataTextHeaders.recordType} must be valid.`,
+  })
+  recordType: RecordType;
+  /**
+   * SIN.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsNumberString(
+    { no_symbols: true },
+    {
+      message: `${DataTextHeaders.sin} must be a valid number.`,
+    },
+  )
+  @Length(SIN_NUMBER_LENGTH, SIN_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.sin} must be of length ${SIN_NUMBER_LENGTH}.`,
+  })
+  sin: string;
+  /**
+   * Application Number.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsNumberString(
+    { no_symbols: true },
+    {
+      message: `${DataTextHeaders.applicationNumber} must be a valid number.`,
+    },
+  )
+  @Length(APPLICATION_NUMBER_LENGTH, APPLICATION_NUMBER_LENGTH, {
```

Comment:
Same https://github.com/bcgov/SIMS/pull/2465/files#r1379325084

Reply:
Same for the `withdrawalDate` ?

Reply:
Same for every single one that is under our control: applicationFound, applicationBelongsToInstitution, studentSINMatch, hasCorrectInstitutionCode, isArchived, applicationStatus, hasPreviouslyBeenWithdrawn, isRecordMatch, withdrawalDate.

Reply:
Does the `applicationStatus` need the `@IsEnum(ApplicationStatus)`?

Reply:
Sorry, missed it. Removed.